library/core/src/option.rs

```diff
@@ -1228,22 +1228,36 @@ fn expect_none_failed(msg: &str, value: &dyn fmt::Debug) -> ! {
 #[stable(feature = "rust1", since = "1.0.0")]
 impl<T: Clone> Clone for Option<T> {
     #[inline]
-    fn clone(&self) -> Self {
+    default fn clone(&self) -> Self {
```

Comment:
Make sure you check how specialization is used for iterators. As I understand it, it's always behind a helper trait so that specialization doesn't leak out into the public API (like I think this would).